File: 2506-count-pairs-of-similar-strings/2506-count-pairs-of-similar-strings.py
class Solution:
    def similarPairs(self, words: List[str]) -> int:
      hash_map = dict()
      ans = 0
      
      for word in words:
        key = tuple(sorted(set(word)))
        hash_map[key] = hash_map.get(key, -1) + 1
        ans += hash_map[key]
      
      return ans
        

File: 2506-count-pairs-of-similar-strings/2506-count-pairs-of-similar-strings.cpp
class Solution {
public:
  int similarPairs(vector<string>& words) {
    unordered_map<int, int> hash_map;
    int key, ans = 0;
    
    for (int i = 0; i < words.size(); ++i) {
      key = 0;
      for (int k = 0; k < words[i].size(); k++) {
        key |= 1 << (words[i][k] - 'a');
      }
      hash_map[key]++;
      ans += (hash_map[key] - 1);
    }
    
    return ans;
  }
};